src/pages/movie-list-page/movie-list-page.jsx

```diff
@@ -0,0 +1,177 @@
+import { useEffect, useState, useRef } from 'react';
+import OptionSelect from '../../components/option-select/option-select';
+import SearchForm from '../../components/search-form/search-form';
+import SortControl from '../../components/sort-control/sort-control';
+import { GENRES, SORT_OPTIONS } from '../../constants';
+import './movie-list-page.css';
+import { getMovieList } from '../../movies.api';
+import MovieTile from '../../components/movie-tile/movie-tile';
+import MovieDetails from '../../components/movie-details/movie-details';
+
+const MOVIES_PER_PAGE = 6;
+
+const MovieListPage = () => {
+  const [movies, setMovies] = useState([]);
+  const [selectedMovie, setSelectedMovie] = useState(null);
+  const [total, setTotal] = useState(0);
+  const [page, setPage] = useState(1);
+  const [genre, setGenre] = useState(null);
+  const [searchQuery, setSearchQueary] = useState('');
+  const [sortBy, setSortBy] = useState(SORT_OPTIONS[0].value || null);
+  const offset = (page - 1) * MOVIES_PER_PAGE;
+  const controllerRef = useRef(null);
+
+  useEffect(() => {
+    // Cancel previous request
+    if (controllerRef.current) {
+      controllerRef.current.abort();
+    }
+
+    const controller = new AbortController();
+    controllerRef.current = controller;
+
+    const params = {
+      sortBy,
+      sortOrder: 'asc', //TODO: Modify Sort Control
+      search: searchQuery,
+      searchBy: searchQuery ? 'title' : null,
+      filter: genre === 'ALL' ? null : genre,
+      offset,
+      limit: MOVIES_PER_PAGE,
+    };
+
+    // Remove empty parameters
+    Object.keys(params).forEach((key) => {
+      if (
+        params[key] === '' ||
+        params[key] === null ||
+        params[key] === undefined
+      ) {
+        delete params[key];
+      }
+    });
+
+    const fetchData = async () => {
+      try {
+        const res = await getMovieList(params, controller.signal);
+        setMovies(res.data.data);
+        setTotal(res.data.totalAmount || 0);
+      } catch (err) {
+        if (err.name === 'CanceledError' || err.code === 'ERR_CANCELED') {
+          console.log('Request canceled:', err.message);
+        } else {
+          console.log('Fetch error:', err.message);
+        }
+      }
+    };
+
+    fetchData();
+
+    return () => {
+      if (controllerRef.current) {
+        controllerRef.current.abort();
+      }
+    };
+  }, [offset, searchQuery, genre, sortBy]);
+
+  const totalPages = Math.ceil(total / MOVIES_PER_PAGE);
+
+  const handleSearch = (query) => {
+    if (query !== searchQuery) {
+      setSearchQueary(query);
+      setPage(1);
+    }
+  };
+
+  const handleGenreSelection = (selection) => {
+    setGenre(selection);
+    setPage(1);
+  };
+
+  const handleSortBy = (selection) => {
+    setSortBy(selection);
+    setPage(1);
+  };
+
+  return (
+    <div className="movie-list-page">
```

Comment:
To me it looks like a cluttered component. It renders a lot of HTML elements, along with other components, conditionals, lists, logos, etc.

It is somewhat hard to read and maintain, I would rather have most of the content separated into more fine-grained components

Reply:
Split movie-list-page to several separate components